script/deploy/1_DeployOFT.s.sol

```diff
@@ -0,0 +1,140 @@
+
+                vm.startBroadcast();
+                currentDeployment.erc20Address = deployL2YnERC20(
+                    implementationSalt,
+                    proxySalt,
+                    baseInput.erc20Name,
+                    baseInput.erc20Symbol,
+                    baseInput.erc20Decimals,
+                    deployer,
+                    currentDeployment.oftAdapterTimelock
+                );
+
+                vm.stopBroadcast();
+
+                console.log("L2ERC20 deployer at: %s", currentDeployment.erc20Address);
+                require(predictedERC20 == currentDeployment.erc20Address, "ERC20 Prediction mismatch");
+            } else {
+                console.log("Already deployed L2ERC20 at: %s", currentDeployment.erc20Address);
+            }
+        }
+        currentDeployment.erc20ProxyAdmin =
+            getTransparentUpgradeableProxyAdminAddress(currentDeployment.erc20Address);
+        currentDeployment.erc20Implementation =
+            getTransparentUpgradeableProxyImplementationAddress(currentDeployment.erc20Address);
+
+        if (!isContract(currentDeployment.oftAdapter)) {
+            console.log("Deploying OFTAdapter");
+
+            bytes32 proxySalt = createOFTAdapterProxySalt(deployer);
+            bytes32 implementationSalt = createOFTAdapterImplementationSalt(deployer);
+
+            address predictedOFTAdapter = CREATE3_FACTORY.getDeployed(deployer, proxySalt);
+            vm.startBroadcast();
+            if (currentDeployment.isL1) {
+                currentDeployment.oftAdapter = deployL1YnOFTAdapter(
+                    implementationSalt,
+                    proxySalt,
+                    currentDeployment.erc20Address,
+                    getData(block.chainid).LZ_ENDPOINT,
+                    deployer,
+                    currentDeployment.oftAdapterTimelock
+                );
+            } else {
+                currentDeployment.oftAdapter = deployL2YnOFTAdapter(
+                    implementationSalt,
+                    proxySalt,
+                    currentDeployment.erc20Address,
+                    getData(block.chainid).LZ_ENDPOINT,
+                    deployer,
+                    currentDeployment.oftAdapterTimelock
+                );
+            }
+            vm.stopBroadcast();
+            console.log("OFTAdapter deployed at: %s", currentDeployment.oftAdapter);
+            require(predictedOFTAdapter == currentDeployment.oftAdapter, "OFT Adapter Prediction mismatch");
+        } else {
+            console.log("Already deployed OFTAdapter at: %s", currentDeployment.oftAdapter);
+        }
+
+        currentDeployment.oftAdapterProxyAdmin =
+            getTransparentUpgradeableProxyAdminAddress(currentDeployment.oftAdapter);
+        currentDeployment.oftAdapterImplementation =
+            getTransparentUpgradeableProxyImplementationAddress(currentDeployment.oftAdapter);
+
+        if (!currentDeployment.isL1) {
```

Comment:
STEP 4: If L2, then grant the necessary roles on the ERC20 to the OFT Adapter and renounce role from deployer